### DISCUSSION TITLE
Problems encountered in offline development

### DISCUSSION THREAD

**Discussion by zzkkui** [eclipse-theia/theia#10368] [Q&A] [answered]
When I was developing and building offline, I encountered this problem. I need to download something from github. How can I solve this problem?
![Image_20211102153823](https://user-images.githubusercontent.com/24466870/139805147-4747c85b-c634-4470-8ac4-3d892dc028ff.png)

thanks

**Comment by msujew** [comment, marked as answer]
Hi @zzkkui,

As the [`postinstall` script of the `vscode-ripgrep`](https://github.com/microsoft/vscode-ripgrep/blob/cefc54056403f76f61c65d40b8868d299810e378/lib/postinstall.js#L65-L69) repo indicates, it won't try to download it, if the appropriate directory already exists on the system. You could trick the download script into thinking it already downloaded it by placing the folder there yourself (hopefully with the correct binary distribution). There's no way that I'm aware of to skip this, since the binary is necessary for the `search-in-workspace` feature to work, as it runs the `ripgrep` binary on the workspace to produce search results.
